src/jsifier.js

```diff
@@ -422,6 +422,8 @@ function JSify(data, functionsOnly) {
     // "Final shape that will be created").
     print('// EMSCRIPTEN_END_FUNCS\n');
 
+    if (ASSERTIONS) print('var ASSERTIONS = true;\n');
+    print(read('arrayUtils.js'));
```

Comment:
don't you need `processMacros(preprocess(` (or maybe just one of those) on what is read, so that it handles the `{{{` stuff? and speaking of which, how is that handled in the html, is that a remaining open question?